tools/proteore_topgo/topGO.xml

```diff
@@ -0,0 +1,268 @@
+        <requirement type="package" version="3.8.2">bioconductor-org.rn.eg.db</requirement>
+        <requirement type="package" version="1.62.0">bioconductor-graph</requirement>
+        <requirement type="package" version="1.46.0">bioconductor-annotationdbi</requirement>
+        <requirement type="package" version="3.8.2">bioconductor-go.db</requirement>
+        <requirement type="package" version="2.36.0">bioconductor-topgo</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:" />
+    </stdio>
+    <command><![CDATA[
+
+  Rscript --vanilla $__tool_directory__/topGO_enrichment.R
+  --inputtype="$inputtype.filetype"
+  --input='$inputtype.genelist'
+
+  #if $inputtype.filetype == "file"
+    --column='$inputtype.column'
+    --header='$inputtype.header'
+  #end if
+
+  --ontology='$ontocat'
+  --option='$option'
+  --threshold='$threshold'
+  --correction='$correction'
+  --textoutput='true'
+  --plot='$plot'
+  --geneuniverse='$geneuniverse'
+  --background="$background_genes.background"
+
+  #if $background_genes.background == "true"
+    --background_genes="$background_genes.inputtype.genelist"
+    --background_input_type="$background_genes.inputtype.filetype"
+    #if $background_genes.inputtype.filetype == "file"
+      --background_header="$background_genes.inputtype.header"
+      --background_column="$background_genes.inputtype.column"
+    #end if
+  #end if
+
+    ]]></command>
+
+    <inputs>
+  <conditional name="inputtype">
+    <param name="filetype" type="select" label="Enter your IDs (Ensembl Gene Id)" help="Copy/paste or from a file">
+      <option value="file" selected="true">Input file containing your IDs</option>
+      <option value="copy_paste">Copy/paste your list of IDs</option>
+    </param>
+    <when value="copy_paste">
+      <param name="genelist" type="text" label="Enter a list of IDs">
+        <sanitizer>
+            <valid initial="string.printable">
+                <remove value="&apos;"/>
+            </valid>
+            <mapping initial="none">
+                <add source="&apos;" target="__sq__"/>
+            </mapping>
+        </sanitizer>
+      </param>
+    </when>
+    <when value="file">
+      <param name="genelist" type="data" format="txt,tabular" label="Select your file" help=""/>
+      <param name="column" type="text" label="Column number of IDs" help="For example, fill in 'c1' if it is the first column, 'c2' if it is the second column and so on">
+        <validator type="regex" message="Please enter a column number, for example: 'c1' for the first column">[c]{0,1}[0-9]+</validator>
+      </param>
+      <param name="header" type="boolean" checked="true" truevalue="true" falsevalue="false" label="Does file contain header?" />
+    </when>
+  </conditional>
+  <conditional name="background_genes">
+    <param name="background" type="boolean" checked="false" truevalue="true" falsevalue="false" label="Use your own background (ID list) ?"/>
```

Comment:
Please do not use booleans in conditionals. See here https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html#booleans

Use a select instead.